wire/Makefile

```diff
@@ -38,10 +38,11 @@ extract-bolt-csv: bolt-precheck
 	@echo '#include <wire/onion_defs.h>' > wire/extracted_onion_wire_csv
 	@$(BOLT_EXTRACT) $(LOCAL_BOLTDIR)/04*.md >> wire/extracted_onion_wire_csv
 
-# Explicit command to add diff of BOLT CSV's to an experimental_csv file
+# Explicit command to add patchfile for BOLT CSV's
 extract-experimental-bolt-csv: bolt-precheck
-	@$(BOLT_EXTRACT) $(LOCAL_BOLTDIR)/0[127]*.md | diff -u wire/extracted_peer_wire_csv - >>wire/extracted_peer_experimental_csv | if [ $$? -lt 0 ];then exit 1;fi
-	@{ echo '#include <wire/onion_defs.h>'; $(BOLT_EXTRACT) $(LOCAL_BOLTDIR)/04*.md; } | diff -u wire/extracted_onion_wire_csv - >>wire/extracted_onion_experimental_csv | if [ $$? -lt 0 ];then exit 1;fi
+	@$(eval BOLT_GIT_SHA := $(shell git -C $(BOLTDIR) log | head -n1 | cut -c 8-16))
```

Comment:
uses the git sha of the bolt-directory

Reply:
`git rev-parse HEAD` ?  But elsewhere we use BOLTVERSION rather than whatever happens to be checked out in $BOLTDIR.  I can see argumetns for both.

The mix of Make and shell here is really doing my head in, too! :boom: